docs/docs/agents/context.md

```diff
@@ -1,61 +1,120 @@
+Context includes *any* data outside the message list that can shape agent behavior or tool execution. This can be:
 
-=== "Via state"
+- Information passed at runtime, like a `user_id` or API credentials.
+- Internal state updated during a multi-step reasoning process.
+- Persistent memory or facts from previous interactions.
 
-    ```python
-    from langgraph.prebuilt.chat_agent_executor import AgentState
+LangGraph provides **three** primary ways to supply context:
 
-    class CustomState(AgentState):
-        # highlight-next-line
-        custom_data: str
+| Type                         | Description                                   | Mutable? | Lifetime                |
+|------------------------------|-----------------------------------------------|----------|-------------------------|
+| **Config**                   | data passed at the start of a run             | ❌        | per run                 |
+| **State**                    | dynamic data that can change during execution | ✅        | per run or conversation |
+| **Long-term Memory (Store)** | data that can be shared between conversations | ✅        | across conversations    |
 
-    agent = create_react_agent(
-        model="anthropic:claude-3-7-sonnet-latest",
-        tools=[...],
-        # highlight-next-line
-        state_schema=CustomState
-    )
+You can use context to:
 
-    agent.invoke({
-        "messages": "hi!",
-        # highlight-next-line
-        "custom_data": "my custom data"
-    })
-    ```
+- Adjust the system prompt the model sees
+- Feed tools with necessary inputs
+- Track facts during an ongoing conversation
 
-=== "Via config"
+## Providing Runtime Context
 
-    ```python
-    agent = create_react_agent(...)
+Use this when you need to inject data into an agent at runtime.
 
-    agent.invoke(
-        {"messages": "hi!"},
-        # highlight-next-line
-        {"configurable": {"custom_data": "my custom data"}}
-    )
-    ```
+### Config (static context)
 
-## Prompt
+Config is for immutable data like user metadata or API keys. Use
+when you have values that don't change mid-run.
 
-To include context in agent's system prompt (for example, user information), you can define a prompt as a function. This function takes the agent state and config and returns a list of messages to send to the chat model:
+Pass information via the `config` argument. The information should be using the "configurable" key, which is reserved key for this purpose.
 
 ```python
+agent.invoke(
+    {"messages": "hi!"},
+    # highlight-next-line
+    config={"configurable": {"user_id": "user_123"}}
+)
+```
+
+### State (mutable context)
+
+State acts as short-term memory during a run. It holds dynamic data that can evolve during execution, such as values derived from tools or LLM outputs.
+
+```python
+class CustomState(AgentState):
+    # highlight-next-line
+    user_name: str
+
+agent.invoke({
```

Comment:
i think we need to show passing `state_schema` to `create_react_agent` for this to work